### PR TITLE
ARROW-7004: [Plasma] Make it possible to bump up object in LRU cache

### PR DESCRIPTION
**Pull request by pcmoritz**
To avoid evicting objects too early, we sometimes want to bump up a number of objects up in the LRU cache. While it would be possible to call Get() on these objects, this can be undesirable, since it is blocking on the objects if they are not available and will make it necessary to call Release() on them.

### CODE REVIEW

File: cpp/src/plasma/eviction_policy.cc [hunk 1]
```diff
@@ -152,6 +156,13 @@ void EvictionPolicy::RemoveObject(const ObjectID& object_id) {
   cache_.Remove(object_id);
 }
 
+void EvictionPolicy::RefreshObjects(const std::vector<ObjectID>& object_ids) {
+  for (const auto& object_id : object_ids) {
+    int64_t size = cache_.Remove(object_id);
+    cache_.Add(object_id, size);
```
Comment (edoakes) [inline cpp/src/plasma/eviction_policy.cc]: Shouldn't we be checking if size != -1 here or does Add check that explicitly?

File: cpp/src/plasma/protocol.cc [hunk 2]
```diff
@@ -780,4 +780,39 @@ Status ReadDataReply(uint8_t* data, size_t size, ObjectID* object_id,
   return Status::OK();
 }
 
+// RefreshLRU messages
```
Comment (edoakes) [inline cpp/src/plasma/protocol.cc]: ### 
```suggestion
// RefreshLRU messages.
```

**Review by edoakes — APPROVED** [review]
Looks good! One thing to note is it would be better to fetch all of the bumped objects at once rather than calling Remove() on each in sequence, but I like the simplicity of this.

File: cpp/src/plasma/eviction_policy.cc (hunk 1 above)
Comment (pcmoritz) [reply cpp/src/plasma/eviction_policy.cc]: Good point, I added the check!

File: cpp/src/plasma/common_generated.h [hunk 3]
```diff
@@ -183,14 +183,14 @@ inline ObjectInfoT *ObjectInfo::UnPack(const flatbuffers::resolver_function_t *_
...
-  { auto _e = digest(); if (_e) _o->digest = _e->str(); }
-  { auto _e = is_deletion(); _o->is_deletion = _e; }
+  { auto _e = object_id(); if (_e) _o->object_id = _e->str(); };
+  { auto _e = data_size(); _o->data_size = _e; };
+  { auto _e = metadata_size(); _o->metadata_size = _e; };
+  { auto _e = ref_count(); _o->ref_count = _e; };
+  { auto _e = create_time(); _o->create_time = _e; };
+  { auto _e = construct_duration(); _o->construct_duration = _e; };
+  { auto _e = digest(); if (_e) _o->digest = _e->str(); };
+  { auto _e = is_deletion(); _o->is_deletion = _e; };
```
Comment (wesm) [inline cpp/src/plasma/common_generated.h]: These semi-colons are not desirable 

https://github.com/google/flatbuffers/issues/5482

can you please remove them?

**Pull request closed without merging**